packages/integration-tests/src/entities/Book.test.ts

```diff
@@ -16,4 +16,35 @@ describe("Book", () => {
     const b1 = em.create(Book, { title: "b1", author: a1 });
     expect(b1.order).toEqual(0);
   });
+
+  it("can change authors", async () => {
+    const em = newEntityManager();
+    // Given an author with a loaded list of books
+    const a1 = em.create(Author, { firstName: "a1" });
+    const b1 = em.create(Book, { title: "b1", author: a1 });
+    // And the book is initially in the author's loaded collection
+    expect(a1.books.get).toEqual([b1]);
+    // When we make a new author and move to the book to it
+    const a2 = em.create(Author, { firstName: "a2" });
+    b1.author.set(a2);
+    // Then both a1 and a2 book collections are correct
+    expect(a1.books.get).toEqual([]);
```

Comment:
Without the fix in this PR, the `books` still had `b1` in it, even though it was supposed to have moved over to `a2`.